src/util/logger/file_log.rs

```diff
@@ -75,23 +68,27 @@ impl RotatingFileLogger {
         })
     }
 
-    /// Opens log file with append mode. Creates a new file if it doesn't exist.
-    fn open(&mut self) -> io::Result<()> {
-        self.file = Some(open_log_file(&self.file_path)?);
-        Ok(())
-    }
-
     fn should_rotate(&mut self) -> bool {
         Utc::now() > self.next_rotation_time
     }
 
     /// Rotates the current file and updates the next rotation time.
     fn rotate(&mut self) -> io::Result<()> {
-        self.close()?;
+        self.flush()?;
+
         let new_path = rotation_file_path_with_timestamp(&self.file_path, &Utc::now());
+        let mut tmp_new_path = new_path.clone().into_os_string();
```

Comment:
Why not just move L85 and L90 to L80? Temp files seem unnecessary.

Reply:
OK fixed in 00b77ea.